appstore/core/models.py

```diff
@@ -1,7 +1,21 @@
+import secrets
 from django.db import models
+from django.utils import timezone
 from django.contrib.auth import get_user_model
+from django.contrib.sessions.models import Session as SessionModel
 from django.core.exceptions import ValidationError
 from django_saml2_auth.user import get_user
+from datetime import timedelta
+from string import ascii_letters, digits, punctuation
+
+UserModel = get_user_model()
+
```

Comment:
This looked odd due to multiple runs through of 256, so I asked cGPT to optimize.    
   # Characters to use for the token
    characters = string.ascii_letters + string.digits

    # Generate a token and check for uniqueness
    while True:
        token = ''.join(secrets.choice(characters) for _ in range(256))
        if not UserIdentityToken.objects.filter(token=token).exists():
            return token